src/LorisInstance.php

```diff
@@ -50,6 +50,36 @@ public function getDatabaseConnection() : \Database
         return $this->DB;
     }
 
+    /**
+     * Return a new database connection to this LORIS instance.
+     *
+     * @return \Database
+     */
+    public function getNewDatabaseConnection() : \Database
```

Comment:
It's useful any place that calls `setBuffering(false)` and then iterates over a query result because the connection can't be re-used until the iteration is finished.

It's the only one now, but there will be others in other PRs.

Reply:
(ie. #9347 is another example that will use it.)

Reply:
Ok yes, thanks. I will make a quick test today ideally.